File: assets/cours/MdP/MdP_demo.c
#include <stdio.h>

/* déclaration */
void is_mult3_liste(int n);
void is_mult3_reste(int n);

/* implémentation */
void is_mult3_liste(int n){
  int L[33] = {3,6,9,12,15,18,21,24,27,30,33,36,39,42,45,48,51,54,57,60,63,66,69,72,75,78,81,84,87,90,93,96,99}; 
  int arrLen = sizeof L;
  
  int isElementPresent = 0;
     
  for (int i = 0; i < arrLen; i++) {
    if (L[i] == n) {
      isElementPresent = 1;
      break;
    }
  }
  
  if (isElementPresent == 1){
    printf("%d est un multiple de 3\n", n);
  }
  else{
    printf("%d n'est pas un multiple de 3\n", n);
  }
}

void is_mult3_reste(int n){
  if (n % 3 == 0){
    printf("%d est un multiple de 3\n", n);
  }
  else{
    printf("%d n'est pas un multiple de 3\n", n);
  }
}

/* Programme principal */

int main() {
  int n=100;
  printf("Programme MdP_demo.c");
  printf("calcul du résultat par la première méthode :\n");
  is_mult3_liste(n);
  printf("calcul du résultat par la seconde méthode :\n");
  is_mult3_reste(n);
  return 0;
}

File: assets/cours/MdP/MdP_demo.py
# -*- coding: utf-8 -*-

# liste des multiples de 3
L_mult3=[3,6,9,12,15,18,21,24,27,30,33,36,39,42,45,48,51,54,57,60,63,66,69,72,75,78,81,84,87,90,93]

def is_mult3_liste(n):
    # implémentation premier algorithme
    if n in L_mult3:
        print("%d est un multiple de 3"%n)
    else:
        print("%d n'est pas un multiple de 3"%n)


def is_mult3_reste(n):
    # implémentation deuxième algorithme
    if n%3==0:
        print("%d est un multiple de 3"%n)
    else:
        print("%d n'est pas un multiple de 3"%n)


### Programme principal

if __name__ == "__main__":
    print("Programme MdP_demo.py")
    n=7
    print("calcul du résultat par la première méthode :")
    is_mult3_liste(n)
    print("calcul du résultat par la seconde méthode :")
    is_mult3_reste(n)
